### PR TITLE
Add GriefPrevention

### PR DESCRIPTION
**Pull request by RoboMWM**
- Provides self-service claims (I changed the claim tool from a gold shovel to a wood shovel so it's more easily accessible.)
- Handles spammers properly via a "softmute" effect, where only they see their own spam. NCP lets spam flood chat, only taking action via kicking when it has reached a certain threshold, while GP silently "softmutes" spam until the player has stopped spamming (or gets themselves kicked for macrospam). Filters gibberish, repeated messages, macrospam, and IP addresses (not domains)
- PvP rules for non-PvP worlds: pet & horse protection, disallow placing fire/lava near players, protects drops on death (can toggle via player and config)
- Smarter ban system which auto-bans players who have the same IP as the recently-banned player. (lasts only a day, as IPs can easily change)

https://www.spigotmc.org/resources/griefprevention.1884

### CODE REVIEW

**Comment by RoboMWM** [comment]
Configured to have minimal impact to anything else. Only exception is if a plotworld plugin creates a world, might need to disable claim function for that world in GP's config. Will PR a change to use GP's spam prevention instead of NCP if this is merged.

**Comment by JamieSinn** [comment]
As said before, is this simply used for claiming land? If so, this really shouldn't be used. A better claim land plugin should. If that is the case, :-1:

**Comment by ghost** [comment]
:-1: No claims needed

**Comment by RoboMWM** [comment]
No - used for PvP grief, spam (although I disabled here unless you want me to disable NCP spam) as well. Will disable claims if not needed

**Comment by ghost** [comment]
PVP Is already disabled, as mentioned on the forums.

**Comment by JamieSinn** [comment]
WG/NCP handles everything that is proposed that this handles.

**Comment by RoboMWM** [comment]
Also handles PvP rules for non-PvP, such as pet & horse protection, not placing fire/lava near players, etc.

Handles spammers better via a "softmute" effect, where they only see their own spam.

**Comment by ctian1** [comment]
This plugin allows players to claim their own land, as opposed to WG

it also offers many other useful anti-grief/admin related features that can be found [here](https://www.spigotmc.org/resources/griefprevention.1884/)

**Comment by RoboMWM** [comment]
I'm not sure if self-serve claims are wanted. As it stands, I have that disabled based on preliminary feedback here.

However, it provides robust spam protection and enforces non-PvP rules such as pet/horse protection and lava/fire placement near players.

**Comment by ctian1** [comment]
Why not use self-serve claims?

**Comment by RoboMWM** [comment]
Not sure.

>A better claim land plugin should [be used].

Curious to know how this is an inferior claim plugin, considering it's been around for 2 years and highly active..

File: plugins/GriefPreventionData/config.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    DeliverManuals: false
+  Spam:
+    Enabled: true
+    LoginCooldownSeconds: 3
+    ChatSlashCommands: /me;/tell;/global;/local;/w;/msg;/r;/t
+    WhisperSlashCommands: /tell;/pm;/r;/w;/whisper;/t;/msg
+    WarningMessage: Please reduce your noise level.  Spammers will be punted.
+    BanOffenders: false
+    BanMessage: Punted for spam.
+    AllowedIpAddresses: 1.2.3.4; 5.6.7.8
```
Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:49]: ...

File: plugins/GriefPreventionData/config.yml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    DeathMessageCooldownSeconds: 60
+  PvP:
+    RulesEnabledInWorld:
+      world: true
+    ProtectFreshSpawns: true
+    PunishLogout: true
+    CombatTimeoutSeconds: 10
+    AllowCombatItemDrop: false
+    BlockedSlashCommands: /home;/vanish;/spawn;/tpa
```
Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:58]: vanish?

File: plugins/GriefPreventionData/config.yml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    ProtectPlayersInLandClaims:
+      PlayerOwnedClaims: true
+      AdministrativeClaims: true
+      AdministrativeSubdivisions: true
+    AllowLavaDumpingNearOtherPlayers: true
```
Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:63]: I thought these would be disabled

Comment (RoboMWM) [inline plugins/GriefPreventionData/config.yml:63]: Only affects PvP-enabled worlds.

File: plugins/GriefPreventionData/config.yml (hunk 2 above)
Comment (RoboMWM) [inline plugins/GriefPreventionData/config.yml:58]: These are part of default config. Can be altered if needed.

File: plugins/GriefPreventionData/config.yml (hunk 3 above)
Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:63]: My bad.

File: plugins/GriefPreventionData/config.yml (hunk 2 above)
Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:58]: I know, I'm saying it should probably be edited

Comment (RoboMWM) [inline plugins/GriefPreventionData/config.yml:58]: Doesn't matter since no PvP anyways.

Comment (ctian1) [inline plugins/GriefPreventionData/config.yml:58]: Should the PvP section just be removed?

Comment (RoboMWM) [inline plugins/GriefPreventionData/config.yml:58]: 'RulesEnabledInWorld' also provides pet/horse protection and flint and steel/lava dumping protection. Plugin will just regen config keys anyways.

**Comment by jeff142** [comment]
:-1:

**Comment by ghost** [comment]
:-1: We don't need claims.

**Comment by RoboMWM** [comment]
Claims are disabled, this isn't the only thing it does...

Re-emphasized antispam in description. NCP does not do much about spam except for extreme cases, and only via a kick. It does not even attempt to mute stuff that is clearly spam and knows is spam because chat antispam is not the primary focus of NCP.

**Comment by md-5** [comment]
Some sort of claiming needed :+1:

**Comment by Maximvdw** [comment]
I like some way of claiming or protecting,  but not griefprevention for protection :-1:

**Comment by RoboMWM** [comment]
What part about it do you not like?

**Comment by TheGrimsey** [comment]
:+1: I like it.

**Comment by Cldfire** [comment]
:-1:
I'd prefer districts to this https://www.spigotmc.org/resources/districts.989/

Very user-friendly, GUI based, great dev.

**Comment by RoboMWM** [comment]
Only thing I see going for that is if people want to micromanage their claims with a chest GUI, that could do (I'd suppose a simple /trust would suffice), and I'll just disable claims from this plugin (as districts does not provide any of the other features mentioned such as anti-spam).

Otherwise, seems to add extra unnecessary things (entrance and exit messages) and lots of economy stuff. Also unknown if more blocks can be acquired other than purchasing (GP adds via non-AFK playtime, and I bumped this up a bit from the default).

**Comment by HappyHarm** [comment]
Maybe someone can create a list with claiming plugins where we can choose about? Then we can better compare them and see which one is the most useful?

**Comment by RoboMWM** [comment]
That'd be helpful because the only feedback I'm getting is "I don't like it" so I don't even know what to address.

**Comment by ctian1** [comment]
IMHO GriefPrevention is a more powerful but easier to use claim plugin than Districts.. you can read about GP [here](https://www.spigotmc.org/resources/griefprevention.1884/)

**Comment by Techcable** [comment]
:+1: some form of claiming is neaded

**Comment by dmulloy2** [comment]
:+1: need claims, GriefPrevention has worked wonderfully for me.

File: plugins/GriefPreventionData/config.yml (hunk 1 above)
Comment (Janmm14) [inline plugins/GriefPreventionData/config.yml:49]: What about adding for example 8.8.8.8 or 12.7.0.0.1?

Comment (RoboMWM) [inline plugins/GriefPreventionData/config.yml:49]: This is just the default whitelist for IP addresses to be allowed in chat (does not block domains).

**Pull request merged**